engine/src/flutter/shell/platform/embedder/embedder.h

```diff
@@ -258,6 +258,40 @@ typedef enum {
   kFlutterSemanticsFlagIsRequired = 1 << 30,
 } FlutterSemanticsFlag;
 
+typedef struct {
+  bool hasCheckedState;
+  bool isChecked;
+  bool isSelected;
+  bool isButton;
+  bool isTextField;
+  bool isFocused;
+  bool hasEnabledState;
+  bool isEnabled;
+  bool isInMutuallyExclusiveGroup;
+  bool isHeader;
+  bool isObscured;
+  bool scopesRoute;
+  bool namesRoute;
+  bool isHidden;
+  bool isImage;
+  bool isLiveRegion;
+  bool hasToggledState;
+  bool isToggled;
+  bool hasImplicitScrolling;
+  bool isMultiline;
+  bool isReadOnly;
+  bool isFocusable;
+  bool isLink;
+  bool isSlider;
+  bool isKeyboardKey;
+  bool isCheckStateMixed;
+  bool hasExpandedState;
+  bool isExpanded;
+  bool hasSelectedState;
+  bool hasRequiredState;
+  bool isRequired;
+} FlutterSemanticsFlags;
```

Comment:
What's an ergonomic/idiomatic way to guarantee that the struct is initialized with all fields set to `false`. AFAIK, C does not generally guarantee that the memory is zeroed out upon initialization. Simply creating a variable of type `FlutterSemanticsFlags` may end up with some random subset of flags set to `true`.

Reply:
looks like we can do FlutterSemanticsFlags{0} to initialize all fields to 0. updated tests.